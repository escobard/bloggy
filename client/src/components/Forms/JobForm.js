import React, { Component } from "react"
import PropTypes from "prop-types";

export default class JobForm extends Component{
	render(){
		return(

		)
	}
}

JobForm.defaultProps = {
	test: "test"
};

JobForm.propTypes = {
	test: PropTypes.string
};