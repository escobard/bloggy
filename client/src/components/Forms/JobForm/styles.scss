@import "../../../styles/variables";
.job-form{
}
	.cancel button, .submit button{
    background-color: $primary-dark !important;
    margin: 24px;
}
.cancel {
    float: left;
    margin: 16px;
}
.submit {
	margin: 16px;
}