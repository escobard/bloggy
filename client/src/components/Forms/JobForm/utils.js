import { emailRe } from "../../../constants"

export default emails =>{
	console.log('EmaiLRE', emailRe)
	let invalidEmails = emails

		// splits the emails string into an array,
		// splitting each string after a comma
		.split(',')

		// maps the array we just created, removes trailing spaces, returns clean array
		.map(email => email.trim())

		// filters the array, checking if any of the strings (emails) within the emails array
		// contains valid characters - aka email@address.com

		// tests to see if our email string contains valid characters
		// tested vs the email regular expression within src/utils
		// this returns all emails that DO NOT match the regex
		// into our array, so we can display them to the user
		.filter(email => !emailRe.test(email))

		// returns invalid emails
		if (invalidEmails.length) {
			return `These emails are invalid: ${invalidEmails}`
		}

}