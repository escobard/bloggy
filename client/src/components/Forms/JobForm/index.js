import React, { Component } from "react";
import PropTypes from "prop-types";
import RaisedButton from "material-ui/RaisedButton";
// this is similar to connect from react-redux
import { reduxForm } from "redux-form";

import InputField from "../Fields/InputField";
// more on this on redux-form.com
// shows a ton of examples that we can use for validation and quick validation helpers
// look at https://redux-form.com/7.3.0/examples/fieldlevelvalidation/ for login user creation val

import { jobFormFields } from "../../../constants";

import styles from "./styles.scss"

class JobForm extends Component {
	renderInputs = () => {
		return jobFormFields.map((field, index) => {
			let { hint, label, name } = field;
			return <InputField key={index} hint={hint} label={label} name={name} />;
		});
	};

	render() {
		console.log(jobFormFields);
		return (
			<div className="job-form">
				<form onSubmit={this.props.handleSubmit(values => console.log(values))}>
					{this.renderInputs()}
				</form>
			</div>
		);
	}
}

JobForm.defaultProps = {
	test: "test"
};

JobForm.propTypes = {
	test: PropTypes.string
};

// validates the form
const validate = (values) =>{
	const errors = {}

	

	// redux form will assume that the form is valid if the
	// errors object is empty
	return errors;
}

export default reduxForm({
	// ES6 deconstruct, grabs the validate function
	// and passes it to the reduxForm HOC
	validate,
	
	// these options customize how we want the form to behave
	form: "jobForm"
})(JobForm);