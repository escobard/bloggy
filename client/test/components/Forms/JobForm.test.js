import React, { Component } from 'react'
import JobForm from '../../../src/components/Forms/JobForm';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('JobForm snapshot', () => {
    const initialState = {output:100}
    const mockStore = configureStore()
    let store,container

    beforeEach(()=>{
        store = mockStore(initialState)
        container = shallow(<JobForm /> )  
    })

    it('+++ render the JobForm component', () => {
       expect(toJson(container)).toMatchSnapshot()
    });

});