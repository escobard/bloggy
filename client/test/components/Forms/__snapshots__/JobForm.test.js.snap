// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobForm snapshot +++ render the JobForm component 1`] = `
<Connect(Form(JobForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="jobForm"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  validate={[Function]}
/>
`;
